skimage/draw/draw.py

```diff
@@ -242,6 +242,9 @@ def polygon_perimeter(r, c, shape=None, clip=False):
 
     Examples
     --------
+    >>> import pytest
```

Comment:
I would prefer using conftest.py to handle skipping tests when matplotlib isn't installed.

Something like this in `conftest.py` should work:
```
try:
    import matplotlib

    has_matplotlib = True
except ImportError:
    has_matplotlib = False

collect_ignore = []

needs_matplotlib = [
    "draw/draw.py",
    "filters/thresholding.py",
    ...
]

if not has_matplotlib:
    collect_ignore += needs_matplotlib
```

Reply:
```
needs_matplotlib = [
    "draw/draw.py",
    "filters/thresholding.py",
    ...
]

collect_ignore = []
try:
    import matplotlib

except ImportError:
    collect_ignore += needs_matplotlib
```

Reply:
Yeah, totally with you on that. That has been bothering me some time for cases with pywt and sklearn as well. 

I didn't know about the solution with `conftest.py` so instead I opted to use doctestplus which I've been meaning to propose for some time, see https://github.com/scikit-image/scikit-image/pull/7289. If that PR get's merged I would update the approach taken here. What do you think about going that direction?

Reply:
Addressed in 4f65ab74af464b01396490bb330a02dbf7dceed1